src/main/java/com/iexec/sms/secret/MeasuredSecretService.java

```diff
@@ -52,18 +53,21 @@ public class MeasuredSecretService {
     private Counter initialSecretsCounter;
     private AtomicLong storedSecretsCount;
     private final Counter addedSecretsSinceStartCounter;
+    private AtomicLong cachedSecretsCount;
+    private final CacheSecretService<?> cacheSecretService;
 
     public MeasuredSecretService(String secretsType,
                                  String metricsPrefix,
                                  Supplier<Long> storedSecretsCountGetter,
                                  ScheduledExecutorService storageMetricsExecutorService,
-                                 int storedSecretsCountPeriod) {
+                                 int storedSecretsCountPeriod,
+                                 CacheSecretService<?> cacheSecretService) {
```

Comment:
We may pass a `Supplier` instead of the whole cache:
```suggestion
                                 Supplier<Long> cachedSecretsCountGetter) {
```

Using such a trick offers multiple benefits:
1. It decouples the caching service and the measuring service
2. The caching service is more protected against misuses if it isn't exposed there
3. It allows to pass a simple `() -> 0` lambda that would avoid to test whether the caching service is null